### ISSUE TITLE
Adding image metadata

### ISSUE THREAD

**Issue by chenaski** [khalyomede/gulp-sharp-responsive#2]
Add the ability to get image metadata when creating an option object.

A common use case is to create images for high resolution screens where we need to get multiple versions of the original
image (1x, 2x, 3x).

It would look something like this:

```js
sharpResponsive({
  formats: ({ width }) => [
    { format: "webp", rename: { suffix: "@3x" } },
    { format: "webp", width: width * 0.5, rename: { suffix: "@2x" } },
    { format: "webp", width: width * 0.25 },
  ]
})
```

Input:

- image.png [4000x2000]

Output:

- image@3x.png [4000x2000]
- image@3x.webp [4000x2000]
- image@2x.png [2000x1000]
- image@2x.webp [2000x1000]
- image.png [1000x500]
- image.webp [1000x500]

**Comment by khalyomede** [comment]
I like this idea! Thanks for suggesting it.

What about having the anonymous function at the key level, so that only the key that need to be computed from the metadata define its anonymous function? Something like:

```node
sharpResponsive({
  formats: [
    { format: "webp", rename: { suffix: "@3x" } },
    { format: "webp", width: (metadata) => metadata.width * 0.5, rename: { suffix: "@2x" } },
    { format: "webp", width: (metadata) => metadata.width * 0.25 },
  ]
})
```

What do you think of this API?

**Comment by chenaski** [comment]
Looks great, I think this version is more readable.

**Comment by khalyomede** [comment]
Published in [v0.3.0](https://github.com/khalyomede/gulp-sharp-responsive/releases/tag/v0.3.0).

**Comment by chenaski** [comment]
Excellent work. Thank you @khalyomede 🙂
